testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jaxrs/integration/cdi/CdiInjectionTestCase.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright The WildFly Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.jboss.as.test.integration.jaxrs.integration.cdi;
+import java.net.URISyntaxException;
+import java.net.URL;
+import jakarta.ws.rs.client.Client;
+import jakarta.ws.rs.client.ClientBuilder;
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.core.UriBuilder;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author <a href="mailto:jperkins@redhat.com">James R. Perkins</a>
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+public class CdiInjectionTestCase {
+
+    @Deployment(testable = false)
+    public static Archive<?> deploy() {
+        return ShrinkWrap.create(WebArchive.class, CdiInjectionTestCase.class.getSimpleName() + ".war")
+                .add(new StringAsset("<beans bean-discovery-mode=\"annotated\"></beans>"), "WEB-INF/beans.xml")
+                .addClasses(CDIApplication.class, CDIProvider.class, CDIResource.class, CDIBean.class, InjectionResource.class);
+    }
+
+
+    private static Client CLIENT;
+
+    @ArquillianResource
+    private URL url;
+
+    @BeforeClass
+    public static void createClient() {
+        CLIENT = ClientBuilder.newClient();
+    }
+
+    @AfterClass
+    public static void closeClient() {
+        if (CLIENT != null) {
+            CLIENT.close();
+        }
+    }
+
+    @Test
+    public void checkApplication() throws Exception {
+        checkResponse("app", CDIApplication.class);
+    }
+
+    @Test
+    public void checkProvider() throws Exception {
+        checkResponse("provider", CDIProvider.class);
+    }
+
+    @Test
+    public void checkResource() throws Exception {
+        checkResponse("resource", CDIResource.class);
+    }
+
+    private void checkResponse(final String path, final Class<?> expectedType) throws Exception {
```

Comment:
Wouldn't this pass even previously as the classes might not be CDI beans but would still work as REST components?
What I mean is, shouldn't you test if those classes are resolvable as CDI beans instead (i.e. injectable, or you could even check their `Bean<?>` metadata for scopes etc).

Reply:
That's a good point. I just copied this test from an old commit and it could probably just run in the container and lookup beans instead of trying to use a resource for them.

Reply:
You could use `Instance` and look those classes up as CDI beans programmatically.
If you want to check their metadata, use `Instance.select(...).getHandle().getBean()`; from there you can see scopes, stereotypes etc. used on that given bean.